File: Civitas-Java/Civitas/src/main/java/civitas/SorpresaPORJUGADOR.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author elena
 */
public class SorpresaPORJUGADOR extends Sorpresa{
    
    SorpresaPORJUGADOR(int valor, String texto){   //Resto de sorpresas
        super(valor, texto);
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
                
        if(jugadorCorrecto(actual, todos)){
            Sorpresa aux, aux1;
            
            informe(actual, todos);
            
            aux = new SorpresaPAGARCOBRAR(valor*-1, "Cobra jugador actual");
            
            for(int i = 0; i < todos.size(); i++){
                if(i != actual)
                    aplicarAJugador(i, todos);
            }            
            
            aux1 = new SorpresaPAGARCOBRAR(valor*(todos.size()-1), "Paga jugador actual");           
            aplicarAJugador(actual, todos);
        }
    }
    
    @Override
    public String toString(){
        return super.toString() + 
               "\nTipo Sorpresa: PORJUGADOR";
    }
}

File: Civitas-Ruby/lib/sorpresa_porjugador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "diario"
require_relative "tipo_sorpresa"
require_relative "mazo_sorpresas"
require_relative "tablero"
require_relative "jugador"

module Civitas
  class SorpresaPorjugador < Sorpresa
    def initialize(valor, texto)
      super(valor, texto)
    end
    
    @Override
    def aplicarajugador(actual, todos)
      if jugadorcorrecto(actual, todos)
        sorpresa = SorpresaPagarcobrar.new(super.valor*-1, 'Cobra el jugador actual')
        
        i = 0
        loop do
          if i == todos.length
            if todos[i] != todos[actual]
              sorpresa.aplicarajugador(i, todos)
            end
          end
          i+=1
        end
        
        sorpresa1 = SorpresaPagarCobrar.new(super.valor*(todos.length - 1), 'Paga jugador actual')
        sorpresa1.aplicarajugador(actual, todos)
      end
    end
    
    @Override
    def to_s
      super.to_s + '\nTipo Sorpresa: PORJUGADOR'
    end    
  end
end
